### PR TITLE
New lighter SCT image class

### PR DESCRIPTION
**Pull request by NathanMolinier**
## Context

To cope with the numerous possible image orientations (RPI, LPI, RSA...) when training a deep learning model, it is recommended to adopt a rigorous behaviour by checking/reorienting the image and its groung truth. However, these checks are not simple and modules such as nibabel do not include those features. 

## Description

The idea behind this new script `image.py` is to be able to manipulate easily images (save, check_orientation, change_orientation, get_non_zero_coordinates etc...) without needing to import the entire [spinalcordtoolbox](https://github.com/spinalcordtoolbox/spinalcordtoolbox).

Therefore, efforts were made to duplicate the SCT [image class](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/image.py#L245) while only keeping most relevant features.

### CODE REVIEW

**Comment by NathanMolinier** [comment]
The class Coordinates was also removed from the code: 
- Now: https://github.com/ivadomed/utilities/pull/29/files#diff-fffd989af66b4f62727ad263a031ddd8726c7ebba5a8ec2d46f1bff8a94ce148R156-R180
- Before: https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/image.py#L565-L589

**Comment by NathanMolinier** [comment]
I'm opened to any comments @naga-karthik @valosekj

File: scripts/image.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+import numpy as np
+import nibabel as nib
+import logging
+from copy import deepcopy
+
+logger = logging.getLogger(__name__)
+
+class Image(object):
+    """
+    Compacted version of the https://github.com/spinalcordtoolbox/spinalcordtoolbox/image.py image module
```
Comment (naga-karthik) [inline scripts/image.py:11]:
```suggestion
    Compact version of SCT's Image Class (https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/image.py#L245)
```

File: scripts/image.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+    Create an object that behaves similarly to nibabel's image object. Useful additions include: dims, change_orientation and getNonZeroCoordinates.
```
Comment (naga-karthik) [inline scripts/image.py:12]: Maybe you could also bring `zeros_like`, `empty_like`, `find_zmin_zmax‎` functions into this light-weight class. Those are also useful

**Review by naga-karthik — COMMENTED** [review]
Sorry the delay! Just tested it - this is a great and much-needed script! I just have suggestions for a few more functions to add. Rest looks good!

**Comment by NathanMolinier** [comment]
No problem for the delay ! I just wanted to make sure that the script will be useful for other people.

I added the functions you mentioned as well as the class `SlicerOneAxis`.

**Comment by naga-karthik** [comment]
Thanks for making the additions, they look good to me! Merging

**Comment by valosekj** [comment]
@NathanMolinier thanks for adding this very useful piece of code! 👍🏻

**Pull request merged**